ample/main.py

```diff
@@ -509,6 +470,29 @@ def setup(self, optd):
         logger.info('All needed programs are found, continuing...')
         return optd
 
+    def setup_workdir(self, argso):
+        # Make a work directory - this way all output goes into this directory
+        if argso['work_dir'] and not argso['restart_pkl']:
+            print('Making a named work directory: %s', argso['work_dir'])
+            try:
+                os.mkdir(argso['work_dir'])
+            except Exception as e:
+                msg = "Cannot create work_dir {0}: {1}".format(argso['work_dir'], e)
+                exit_util.exit_error(msg, sys.exc_info()[2])
+
+        if not argso['work_dir']:
+            if not os.path.exists(argso['run_dir']):
+                msg = 'Cannot find run directory: {0}'.format(argso['run_dir'])
+                exit_util.exit_error(msg, sys.exc_info()[2])
+            if bool(argso['rvapi_document']):
+                # With JSCOFE we run in the run directory
+                argso['work_dir'] = argso['run_dir']
+            else:
+                print('Making a run directory: ' 'checking for previous runs...')
```

Comment:
logging

Reply:
also weird single quote style going on, not sure this even works

Reply:
Logging hasn't been set up at this point so we can't use it. I've made that clear in the docstring.

Reply:
that's fine, still doesn't explain the weird single-quote stuff going on there.

Reply:
Oh - I hadn't even noticed that! Good spot.